subsys/net/ip/dhcpv4.c

```diff
@@ -1069,11 +1072,40 @@ static enum net_verdict net_dhcpv4_input(struct net_conn *conn,
 	return NET_DROP;
 }
 
+static void iface_handler(struct net_mgmt_event_callback *cb,
+			  u32_t mgmt_event, struct net_if *iface)
+{
+	if (iface->config.dhcpv4.state == NET_DHCPV4_INIT) {
```

Comment:
read my comment below about k_delayed_work_submit(), I think you don't need that test then